### PR TITLE
Update to rocket 0.5

### PR DESCRIPTION
**Pull request by bash**

### CODE REVIEW

File: src/server.rs [hunk 1]
```diff
@@ -32,66 +35,54 @@ impl RocketServer {
...
+    async fn run(self: Box<Self>, address: IpAddr, port: u16) -> Result<(), Box<dyn Error>> {
+        let config = Config { address, port, ..Default::default() };
 
-        rocket::custom(config)
+        _ = rocket::custom(config)
             .mount("/", routes![search, item_by_slug])
             .manage(self.search_engine)
-            .launch();
+            .launch()
+            .await?;
```
Comment (Mafii) [inline src/server.rs:47]: why not return this?

Comment (Mafii) [reply src/server.rs:47]: Instead of Ok(())

Comment (bash) [reply src/server.rs:47]: `launch()` returns a Result containing a `Rocket` instance, our Server trait requires that we return a unit.

Comment (janhohenheim) [reply src/server.rs:51]: Could do `[...].await.ok().map(|_|())`. I don't think it's actually better, just wanna point it out.

**Review by janhohenheim — APPROVED** [review]

**Pull request merged**